libs/wingc/src/ast.rs

```diff
@@ -702,6 +702,8 @@ pub enum BinaryOperator {
 	LogicalAnd,
 	LogicalOr,
 	UnwrapOr,
+	SubtractAssignment,
+	AddAssignment,
```

Comment:
I think this shouldn't be a `BinaryOperator`. It should be implemented as a new type of assignment statement.
I'd change `StmtKind::Assignment` so it includes a new enum field `AssignmentKind` of a type 'Assign/AssignIncr/AssignDecr`. Then in _type_check.rs_ for the new assignment types we need to validate not only that the source and target expressions are of the same type, but that they are also numbers.

Reply:
Thank you for the review and advice, I will try to change it according to it.

Reply:
I changed it according to your advice, `StmtKind::Assignment` now includes a new enum field to specify the assignment type, the type check is also updated.